### ISSUE TITLE
bounds values set wrongly after calling ``.select`` on ``Image`` object

### ISSUE THREAD

**Issue by RafalSkolasinski** [ioam/holoviews#702]
When `.select` method is called on `Image` object it looks like `bounds` of newly created image are set to the input of `.select` method. I found that in following case:

``` python
import numpy as np
import holoviews
from holoviews import Image, Points
holoviews.notebook_extension()
```

``` python
side = 5
region=(-side,-side,side,side)

def coords(bounds,samples):
    l,b,r,t=bounds
    hc=side/samples
    return np.meshgrid(np.linspace(l+hc,r-hc,samples),
                       np.linspace(b+hc,t-hc,samples))

x,y = coords(region, 5)
z = np.sin(x**2+y**2)

img1 = Image(z, group="Function", label="Sine", bounds=region) 
img2 = img1.clone(extents=(-2, -2, 2, 2))
img3 = img1.select(x=(-2,2), y=(-2,2))
```

``` python
>>> print(img1[0.95, 0], img2[0.95, 0], img3[0.95, 0])
0.0 0.0 -0.756802495308
```

``` python
%%opts Image [situate_axes=True] {+axiswise}
img1*Points([(0.9,0)]) + img2*Points([(0.9,0)]) + img3*Points([(0.9,0)])
```

![2016_06_02_15 01 09](https://cloud.githubusercontent.com/assets/10928117/15745587/db75b91c-28d2-11e6-8e9f-f36d6a015680.png)

it leads to confusion when accessing value at the same coordinates of the initial and new object.
Maybe bounds of a new object should be selected to the edge of edge pixels that got selected?

**Comment by philippjfr** [comment]
Thanks @RafalSkolasinski. This a pretty major and annoying bug and will have to be addressed.

**Comment by jlstevens** [comment]
I've assigned the milestone to 1.6.0 as I agree this is an important bug to address. I would consider moving it to a minor,  post 1.5 release if we decide we are going to have one of those (seems likely!).

**Comment by RafalSkolasinski** [comment]
Hmm... maybe it would be good idea to make
```python
[situate_axes=True]
```
a default setting?

**Comment by philippjfr** [comment]
This turned out to be a pretty important bug which actually caused some distortions in a plot in my thesis. I've finally figured it out in #994, which will hopefully make it into the 1.7 release.

**Comment by philippjfr** [comment]
This has now finally been **finally** been fixed in #994, which was just merged.

```python
>>> print(img1[0.95, 0], img2[0.95, 0], img3[0.95, 0])
(0.0, 0.0, 0.0)
```

```python
%%opts Image [situate_axes=True] {+axiswise}
img1*Points([(0.9,0)]) + img2*Points([(0.9,0)]) + img3*Points([(0.9,0)])
```

![image](https://cloud.githubusercontent.com/assets/1550771/24830241/a6c16680-1c79-11e7-9f5a-9910e5bb44df.png)
